### PR TITLE
#[UIC-1121] Enable Unit test Execution on CI environment for Frond-end(Javascript)

### PR DESCRIPTION
**Pull request by priyanka-singh-guavus**
1.Runs unit test on CI and generated a coverage report for front-end (javascript).

2.The coverage report is generated through jest.

3.In Jenkinsfile folder, the path for this report is:
 HTML_REPORT = 'superset/assets/output/coverage/jest/lcov-report/'

### CODE REVIEW

File: superset/assets/js_build.sh [hunk 1]
```diff
@@ -20,6 +20,7 @@ cd "$(dirname "$0")"
 npm --version
 node --version
 time npm ci
-time npm run lint
-time npm run cover  # this also runs the tests, so no need to 'npm run test'
+#time npm run lint
+echo    "npm run test"
```
Comment (arpit-agarwal) [inline superset/assets/js_build.sh]: Remove echo statement

Comment (arpit-agarwal) [reply superset/assets/js_build.sh]: I don't see a command to invoke this file.
Can we segregate commands as per stages and remove empty stage

File: superset/assets/package.json [hunk 2]
```diff
@@ -21,6 +22,12 @@
     "cypress": "cypress",
     "cypress-debug": "cypress open --config watchForFileChanges=true"
   },
+  "jest": {
+    "coverageReporters": [
+      "text",
+      "cobertura"
```
Comment (bipinsoniguavus) [inline superset/assets/package.json]: I guess, we are not generating `cobertura` report now, if its HTML report then we should remove "cobertura" from here

File: superset/assets/package.json [hunk 3]
```diff
@@ -8,8 +8,9 @@
     "test": "spec"
   },
   "scripts": {
+    "start": "webpack",
```
Comment (bipinsoniguavus) [inline superset/assets/package.json]: What is the intent of adding this script here? Are we using it in any stages?

File: Jenkinsfile [hunk 4]
```diff
@@ -63,6 +63,11 @@ pipeline {
         echo "Run Commmands to trigger build"
       }
     }
+    stage("Code Coverage of JavaScript files") {
```
Comment (bipinsoniguavus) [inline Jenkinsfile]: I guess @arpit-agarwal is mentioned about this stage. Please confirm

File: superset/assets/package.json (hunk 3 above)
Comment (priyanka-singh-guavus) [reply superset/assets/package.json]: Here, we used jest with webpack to manage assets, styles, and compilation.

File: superset/assets/js_build.sh (hunk 1 above)
Comment (priyanka-singh-guavus) [reply superset/assets/js_build.sh]: Removed echo statement and and empty stage also.

File: superset/assets/package.json (hunk 2 above)
Comment (priyanka-singh-guavus) [reply superset/assets/package.json]: Removed "cobertura".

File: superset/assets/package.json (hunk 3 above)
Comment (priyanka-singh-guavus) [reply superset/assets/package.json]: Removed start:"webpack".

File: Jenkinsfile (hunk 4 above)
Comment (priyanka-singh-guavus) [reply Jenkinsfile]: At this stage sh "tox -e javascript" shell script is added.

File: Jenkinsfile [hunk 5]
```diff
@@ -66,6 +66,7 @@ pipeline {
     stage("Code Coverage of JavaScript files") {
       steps {
         echo "Run Commmands to get code coverage of JavaScript Files"
```
Comment (arpit-agarwal) [inline Jenkinsfile]: Update echo to say "Running" rather than "Run"

File: superset/assets/package.json [hunk 6]
```diff
@@ -24,8 +23,7 @@
   },
   "jest": {
     "coverageReporters": [
-      "text",
-      "cobertura"
+      "text"
     ]
```
Comment (arpit-agarwal) [inline superset/assets/package.json]: I think we don't need "jest" section as we are using defaults

File: Jenkinsfile (hunk 5 above)
Comment (priyanka-singh-guavus) [reply Jenkinsfile]: Replaced run as running

File: superset/assets/package.json (hunk 6 above)
Comment (priyanka-singh-guavus) [reply superset/assets/package.json]: Removed these steps.

**Pull request merged**